query/src/main/scala/hmda/query/projections/institutions/InstitutionView.scala

```diff
@@ -0,0 +1,101 @@
+package hmda.query.projections.institutions
+
+import akka.actor.{ ActorRef, ActorSystem, Props }
+import akka.persistence.{ RecoveryCompleted, SnapshotOffer }
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Sink
+import hmda.model.institution.Agency.CFPB
+import hmda.model.institution.InstitutionType.Bank
+import hmda.model.institution.{ Inactive, Institution }
+import hmda.persistence.messages.CommonMessages.{ Command, Event, GetState, Shutdown }
+import hmda.persistence.messages.events.institutions.InstitutionEvents._
+import hmda.persistence.model.HmdaPersistentActor
+import hmda.persistence.processing.HmdaQuery._
+
+object InstitutionView {
+
+  val name = "institutions-view"
+
+  case class GetInstitutionById(institutionId: String) extends Command
+  case class GetInstitutionsById(ids: List[String]) extends Command
+  case class LastProcessedEventOffset(seqNr: Long)
+  case object StreamCompleted
+
+  def props(): Props = Props(new InstitutionView)
+
+  def createInstitutionQuery(system: ActorSystem): ActorRef = {
+    system.actorOf(InstitutionView.props(), "institutions-view")
+  }
+
+  case class InstitutionViewState(institutions: Set[Institution] = Set.empty[Institution], seqNr: Long = 0L) {
+    def updated(event: Event): InstitutionViewState = {
+      event match {
+        case InstitutionCreated(i) =>
+          InstitutionViewState(institutions + i, seqNr)
+        case InstitutionModified(i) =>
+          val others = institutions.filterNot(_.id == i.id)
+          InstitutionViewState(others + i, seqNr)
+      }
+    }
+  }
+
+}
+
+class InstitutionView extends HmdaPersistentActor {
```

Comment:
That's how it is in this PR, yes. So `InstitutionPersistence` is exclusively concerned with writing stuff. `InstitutionView` reads created / modified events as they get to the journal and updates state accordingly and exposes the current state through the API. This part was previously done also in `InstitutionPersistence`, now it's separated. At the same time, it sends these events to `InstitutionDBProjection`, which stores the current state in a relational database. At this point I don't know if we will be able to serve current institution data from memory. I _think_ so, but I am not sure. At the same time, that is not enough to satisfy some of the panel and edits requirements. We need to know what the panel looks like every year. So a SQL view (filtering by year) on that SQL table makes a lot of sense. The SQL table can also serve the API if it gets too big to fit in memory in the future (in that case, the view would just be responsible for reading events from journal and streaming them to consumers). Having this data in a relational database will also help with analytics in the future.